R/g.report.part5.R

```diff
@@ -113,17 +133,24 @@ g.report.part5 = function(metadatadir = c(), f0 = c(), f1 = c(), loglocation = c
     outputfinal = as.data.frame(do.call(rbind,
                                         lapply(fnames.ms5[f0:f1], myfun, expectedCols)),
                                 stringsAsFactors = FALSE)
-    # order data.frame
-    outputfinal$window_number = as.numeric(gsub(" ", "", outputfinal$window_number))
-    outputfinal = outputfinal[order(outputfinal$filename, outputfinal$window, outputfinal$window_number),]
     
     # Find columns filled with missing values
     cut = which(sapply(outputfinal, function(x) all(x == "")) == TRUE) 
     if (length(cut) > 0) {
       outputfinal = outputfinal[,-cut]
     }
+    
+    # order data.frame
+    outputfinal$window_number = as.numeric(gsub(" ", "", outputfinal$window_number))
+    # outputfinal_segments = outputfinal[which(!outputfinal$window %in% c("MM", "WW")),]
+    # outputfinal = outputfinal[which(outputfinal$window %in% c("MM", "WW")),]
+    outputfinal = outputfinal[order(outputfinal$filename, outputfinal$window, outputfinal$window_number),]
+    
     # split results to different spreadsheets in order to minimize individual filesize and to ease organising dataset
     uwi = as.character(unique(outputfinal$window))
+    if (!all(uwi %in% c("MM", "WW"))) {
```

Comment:
Does this mean that if user only specifies WW segments will also be added?
Should this not be limited to a check of whether MM is present?

Reply:
The object `uwi` is extracted from the part 5 meta data (`outputfinal$window`). With `qwindow = NULL`, uwi would be either `"MM"`, `"WW"`, or `c("MM", "WW")`. In that case, this if statement would be FALSE and would not run the next line.

Otherwise, if `qwindow != NULL`, `outputfinal$window` would additionally contain the names of the segments. For example, it could be `uwi = c("MM", "WW", "segment1", "segment2", segment3")`. However we only need an extra `uwi` beyond "MM" and "WW", to run a third iteration in the loop to calculate the `qwindow` segments. This is done by redefining the segment names in the object `uwi` by the word "Segments". Then, there would be an iteration for "MM" and/or "WW" (as provided by the user) and an extra iteration for "Segments".

Reply:
In the next line:
`uwi = c(uwi[uwi %in% c("MM", "WW")], "Segments")`

Where `uwi[uwi %in% c("MM", "WW")]` makes that only the user-selection regarding "MM" and "WW" is taken. So, "WW" would not be added if not provided by the user.